app/assets/javascripts/to_do_items.coffee

```diff
@@ -40,14 +41,56 @@ class @TodoItem
   detectCancelFormButton = (id) ->
     $("#cancel-item-button-#{id}")
 
+  extractId = (str) ->
+    str.split('-').pop()
+
   removeCallout = (listId) ->
     detectList(listId).find('div.bs-callout').remove()
 
+  saveOnClick = (item) ->
+    $(item).click () ->
+      updateItem extractId(item['id']), item['value']
+
+  markAsDone = (id, listId) ->
+    completeList = detectList(listId).find('ul.complete')
+    detectItem(id).appendTo completeList
+
+  unmark = (id, listId) ->
+    incompleteList = detectList(listId).find('ul.incomplete')
+    detectItem(id).appendTo incompleteList
+
+  cancelChecking = (id) ->
+    currentCheckbox = detectItem(id).find("input[type='checkbox']")
+    currentCheckbox.prop "checked", !currentCheckbox.prop("checked")
+
+  showCheckAlert = (listId) ->
+    alertBox = detectList(listId).find('div.alert')
+    alertBox.fadeIn 'slow'
+    alertBox.delay 2000
+    alertBox.fadeOut 'slow'
```

Comment:
@vlasiak, can we achieve a delay using just `fadeOut` http://api.jquery.com/fadeout/ without `delay` method?